### ISSUE TITLE
Blue/Red War Fuel

### ISSUE THREAD

**Issue by lactoseti** [TrinityCore/TrinityCore#23053]
**Description:**
Blue war fuel (item=46765) and red war fuel appears to cast a spell on the warbot/battlebot, but no buff (filled up) ever goes on it, and the model doesn't change to the respective blue or red bot.

**Current behaviour:**

npc=34587 is the warbot pet, given by item=46767

You can use item 46765 or 46766 when this pet is out; it will cast a spell with lightning, but no buff ever goes on the pet.

**Expected behaviour:**

Pet should have buff.

**Steps to reproduce the problem:**
Pull out pet, use item 46765 or 46766

Branch(es): 3.3.5
TC rev. hash/commit: [58c5dfa](https://github.com/TrinityCore/TrinityCore/commit/58c5dfac1d8d314b9243fde52466b3c933fb7d7f)
Os: windows 2016 server

**Comment by Killyana** [comment]
Many spells must be used by the script: https://woehead.way-of-elendil.fr/?search=Warbot#spells

**Comment by lactoseti** [comment]
When bots of two different colors come within a distance of each other they need to fight.  I guess some of those spells are needed for the fight?

**Comment by illfated** [comment]
Hmm, isn't the Pet Battles (introduced in Cataclysm) necessary for non-combat pet fights to work?
Expanded: Pet Battle System: (Mists of Pandaria) Patch 5.0.4 (2012-08-28): Added.

**Comment by offl** [comment]
According to spell effects, very likely it's an imitation of fight. I mean they will not be in combat.

**Comment by lactoseti** [comment]
you're right; it's an imitation; it was in game in 2009--in more modern patches (5.x onward) you can buy fuel in dalaran, works the same these days

**Comment by lactoseti** [comment]
I certainly don't mind taking a stab at it myself, but I'm pretty new--maybe could use a general pointer--I've added some custom scripts (for example, I made one to make Landro work and wrote some simple code to let accounts have flags for collector's editions, etc., sends out emails on login to the characters)--can anyone point me to the basic location to put this script for the battlebots?

**Comment by illfated** [comment]
I could be mistaken, but I think it more or less depends on what kind of scripts they are.

If they are purely spell scripts, check out the non-player, non-class scripts in
https://github.com/TrinityCore/TrinityCore/tree/3.3.5/src/server/scripts/Spells ,
whereas if the scripts are defined as item or NPC scripts, they likely belong in the
https://github.com/TrinityCore/TrinityCore/tree/3.3.5/src/server/scripts/World
category.
